include/ccf/service/node_info.h

```diff
@@ -69,6 +69,13 @@ namespace ccf
      * node identity in `public_key` field. Service-endorsed certificate is
      * recorded in "public:ccf.nodes.endorsed_certificates" table */
     std::optional<crypto::Pem> cert = std::nullopt;
+
+    /** Commit state for Retired state
+     *
+     * Introduced during 2.x (2.0.5), and so optional for backward
+     * compatibility.
+     */
+    bool retired_committed = false;
```

Comment:
I thought we once said (probably not documented!) that all added fields going forward should be `std::optional`?

Reply:
It's not strictly speaking necessary if we provide a default value, which we do here, and if that default value is logically correct in the cases where the serialised form does not have the field present (which is the case here). Check with @eddyashton. But in general, yes.

Reply:
I vaguely remember a discussion following https://github.com/microsoft/CCF/discussions/2916 where we said we wanted `std::optional` for new fields to make sure that it was clear for programmers to know that the value may not exist in the ledger. I'm not too fussed either way, but I think we should strive for consistency and we've gone the `std::optional` way so far.

Reply:
From that discussion:

> The new field will either have to be defined as i) std::optional or ii) a non-std::optional type, with a sensible default value.

This falls under ii) as far I can tell? I would argue false is a sensible default here? I'm fine changing this, I just don't see much of a benefit (consistency is dubious, most of our fields aren't optional, users already have to deal with the fact there's a mix).

Reply:
Ok, so after discussion, I will:

1. Keep this as it is for now
2. Look separately at changing serialisation to always serialise non-std::nullopt values (#4074)